bridge/client/app/_components/ktb-evaluation-badge/ktb-evaluation-badge.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, Input, NgZone, OnDestroy, TemplateRef, ViewChild } from '@angular/core';
+import { Trace } from '../../_models/trace';
+import { takeUntil } from 'rxjs/operators';
+import { DtOverlay, DtOverlayConfig, DtOverlayRef } from '@dynatrace/barista-components/overlay';
+import { Subject, Subscription } from 'rxjs';
+import { EvaluationBadgeVariant, IEvaluationBadgeState } from './ktb-evaluation-badge.utils';
+
+@Component({
+  selector: 'ktb-evaluation-badge',
+  templateUrl: './ktb-evaluation-badge.component.html',
+  styleUrls: ['./ktb-evaluation-badge.component.scss'],
+})
+export class KtbEvaluationBadgeComponent implements OnDestroy {
+  private readonly unsubscribe$ = new Subject<void>();
+  private overlayRef?: DtOverlayRef<unknown>;
+  private updateOverlayPositionSubscription = Subscription.EMPTY;
+  public TraceClass = Trace;
+  public EvaluationBadgeFillState = EvaluationBadgeVariant;
+  public overlayConfig: DtOverlayConfig = {
+    pinnable: true,
+  };
+
+  @ViewChild('overlay', { static: true, read: TemplateRef }) overlayTemplate?: TemplateRef<unknown>;
+
+  @Input() overlayDisabled = true;
+  @Input() loading = false;
+  @Input() evaluation?: Trace;
+  @Input() evaluationState: Partial<IEvaluationBadgeState> = {
+    isError: false,
+    isSuccess: false,
+    isWarning: false,
+    fillState: EvaluationBadgeVariant.FILL,
+  };
+
+  constructor(private ngZone: NgZone, private _dtOverlay: DtOverlay) {}
+
+  public showEvaluationOverlay(event: MouseEvent, data?: Trace): void {
+    if (!this.overlayDisabled && this.overlayTemplate && data) {
+      this.overlayRef = this._dtOverlay.create(event, this.overlayTemplate, { ...this.overlayConfig, data });
+      this.updateEvaluationOverlayPosition();
+    }
+  }
+
+  private updateEvaluationOverlayPosition(): void {
+    this.updateOverlayPositionSubscription = this.ngZone.onMicrotaskEmpty
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(() => {
+        this.overlayRef?.updatePosition();
+        // if the content of the overlay changed after initialization the position stayed the same
+      });
+  }
+
+  public hideEvaluationOverlay(): void {
+    if (this.overlayRef) {
+      this._dtOverlay.dismiss();
+      this.updateOverlayPositionSubscription.unsubscribe();
+      this.overlayRef = undefined;
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+}
```

Comment:
Can we only use one unsubscription method? If we keep the subscription anyway we might as well do it like this:

```suggestion
  public ngOnDestroy(): void {
    this.updateOverlayPositionSubscription.unsubscribe();
  }
```

No `unsubscribe$` observable needed, no `.pipe(takeUntil(this.unsubscribe$))` necessary.

Alternatively you could also do the following (and not use the subscription at all):

```
  public hideEvaluationOverlay(): void {
    if (this.overlayRef) {
      this._dtOverlay.dismiss();
      this.unsubscribe$.next(); // <--- 
      this.overlayRef = undefined;
    }
  }
```

Reply:
done